driver-sync/src/test/functional/com/mongodb/client/AtlasDataLakeTest.java

```diff
@@ -0,0 +1,337 @@
+            if (e instanceof MongoCommandException) {
+                assertEquals(expectedErrorCodeName, ((MongoCommandException) e).getErrorCodeName());
+                passedAssertion = true;
+            } else if (e instanceof MongoWriteConcernException) {
+                assertEquals(expectedErrorCodeName, ((MongoWriteConcernException) e).getWriteConcernError().getCodeName());
+                passedAssertion = true;
+            }
+        }
+        return passedAssertion;
+    }
+
+    private String getErrorContainsField(final BsonValue expectedResult) {
+        return getErrorField(expectedResult, "errorContains");
+    }
+
+    private String getErrorCodeNameField(final BsonValue expectedResult) {
+        return getErrorField(expectedResult, "errorCodeName");
+    }
+
+    private String getErrorField(final BsonValue expectedResult, final String key) {
+        if (hasErrorField(expectedResult, key)) {
+            return expectedResult.asDocument().getString(key).getValue();
+        } else {
+            return "";
+        }
+    }
+
+    private boolean hasErrorContainsField(final BsonValue expectedResult) {
+        return hasErrorField(expectedResult, "errorContains");
+    }
+
+    private boolean hasErrorCodeNameField(final BsonValue expectedResult) {
+        return hasErrorField(expectedResult, "errorCodeName");
+    }
+
+    private boolean hasErrorLabelsContainField(final BsonValue expectedResult) {
+        return hasErrorField(expectedResult, "errorLabelsContain");
+    }
+
+    private boolean hasErrorField(final BsonValue expectedResult, final String key) {
+        return expectedResult != null && expectedResult.isDocument() && expectedResult.asDocument().containsKey(key);
+    }
+
+    private List<String> getErrorLabelsContainField(final BsonValue expectedResult) {
+        return getListOfStringsFromBsonArrays(expectedResult.asDocument(), "errorLabelsContain");
+    }
+
+    private List<String> getListOfStringsFromBsonArrays(final BsonDocument expectedResult, final String arrayFieldName) {
+        List<String> errorLabelContainsList = new ArrayList<String>();
+        for (BsonValue cur : expectedResult.asDocument().getArray(arrayFieldName)) {
+            errorLabelContainsList.add(cur.asString().getValue());
+        }
+        return errorLabelContainsList;
+    }
+
+    private boolean hasErrorLabelsOmitField(final BsonValue expectedResult) {
+        return hasErrorField(expectedResult, "errorLabelsOmit");
+    }
+
+    private List<String> getErrorLabelsOmitField(final BsonValue expectedResult) {
+        return getListOfStringsFromBsonArrays(expectedResult.asDocument(), "errorLabelsOmit");
+    }
+
+    private ConnectionString getADLConnectionString() {
+        // NOTE: create a system property for this value
+        return new ConnectionString("mongodb://mhuser:pencil@localhost");
+    }
+
+    private void testInvalidUri(final String uri) {
```

Comment:
This doesn't make sense for a Data Lake test.  There is nothing specific to ADL in the ConnectionString class, and ConnectionString itself doesn't actually try to connect to the server

Is the real intention to test connectivity?  If so, it should be more like com.mongodb.client.ConnectivityTest, where we actually create a MongoClient from the URI and run a command, asserting that the command succeeds.